### ISSUE TITLE
NVDA reads <ul> with 4 items as having 5 items

### ISSUE THREAD

**Issue by Epic-Third-Party-Bug-Reporting** [nvaccess/nvda#16905]
### Steps to reproduce:

1. Go to [https://codepen.io/Epic-Third-Party-Bug-Reporting/pen/rNEBZyG](https://codepen.io/Epic-Third-Party-Bug-Reporting/pen/rNEBZyG)
2. In the iframe, arrow down to the list with four <li> items and arrow through each item
3. Observe that the list is said to have five items despite only having four

### Actual behavior:
frame  list with 5 items List Header 1
bullet Item 1
bullet Item 2
bullet Item 3
bullet Item 4
out of list

### Expected behavior:
frame  list with **4** items List Header 1
bullet Item 1
bullet Item 2
bullet Item 3
bullet Item 4
out of list

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2023.1.0.27913

#### Windows version:
Edition: Windows 11 Enterprise
Version: 23H2

#### Name and version of other software in use when reproducing the issue:
Chrome Version 126.0.6478.183 (Official Build) (64-bit)

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by josephsl** [comment]
Hi, can you test with: NVDA 2024.2, Edge 126, and Firefox 128? That way we can figure out if this is a rendering issue or NVDA is getting confused about item count. Thanks.

**Comment by thgcode** [comment]
I think NVDA is considering all the list's child elements as list items, even those that are not `<li>` elements, this can be tested by creating a `<p>` inside a `<ul>` for example and NVDA will report the `<p>` on the count. Would the correct approach in this case be to introduce another attribute on the virtual buffer that only considers list item count?

**Comment by Adriani90** [comment]
I agree a redesign of list item count needs to be investigated. E.g. for definition lists and description lists, NVDA counts definitions ```<dd>``` or descriptions of terms as separate list items which should not be the case.

@thgcode I think your approach would be the right direction, but not only for ```<li>``` elements, but also for ```<dt>``` elements.

**Comment by Adriani90** [comment]
cc: @SaschaCowley
